src/unsatcores/UnsatCoreBuilder.cc

```diff
@@ -113,6 +113,14 @@ void UnsatCoreBuilder::minimize() {
         return;
     }
 
+    // We need to minimize `namedTerms` which could have been masked by already eliminated non-named terms
+    auto const & termNames = solver.getTermNames();
+    // TODO: make more efficient
+    for (PTRef term : solver.getCurrentAssertionsView()) {
+        if (termNames.contains(term)) { continue; }
+        hiddenTerms.push(term);
```

Comment:
Should we `clear` `hiddenTerms` beforehand?

Reply:
Good point!